frame/democracy/src/lib.rs

```diff
@@ -14,7 +14,122 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Democratic system: Handles administration of general stakeholder voting.
+//! # Democracy Pallet
+//! 
+//! - [`democracy::Trait`](./trait.Trait.html)
+//! - [`Call`](./enum.Call.html)
+//!
+//! ## Overview
+//! 
+//! The Democracy pallet handles the administration of general stakeholder voting.
+//!
+//! There are two different queues that a proposal can be added to before it
+//! becomes a referendum, 1) the proposal queue consisting of all public proposals
+//! and 2) the external queue consisting of a single proposal that originates
+//! from one of the _external_ origins (such as a collective group).
+//!
+//! Every launch period, a proposal will be taken from either the proposal queue
+//! or the external queue in turn. The taken proposal will become a referendum
+//! that can be voted on by any token holder in the system. The voting system
+//! uses time-lock voting by allowing the token holder to set their _conviction_
+//! behind a vote. The conviction will dictate the length of time the tokens
+//! will be locked, as well as the multiplier that scales the vote power.
+//!
+//! ### Terminology
+//!
+//! - **Conviction:** A multiplier that is added to a vote based on the length
+//!   of time the voter has opted to lock their tokens. Longer lock-up periods
+//!   grant a stronger vote.
+//! - **Vote:** A value that can either be in approval ("Aye") or rejection ("Nay")
+//!   of a particular referendum.
+//! - **Proposal:** A submission to the chain that represents an action that is 
+//!   being suggested for the system to adopt.
+//! - **Referendum:** A proposal that is in the process of being voted on for 
+//!   either acceptance or rejection as a change to the system.
+//! - **Proxy:** An account that votes on behalf of a separate "Stash" account
+//!   that holds the funds.
+//! - **Delegation:** Granting your voting power to the decisions of another account.
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Functions
+//!
+//! #### Public
+//!
+//! These calls can be made from any externally held account capable of creating
+//! a signed extrinsic.
+//!
+//! - `propose` - The submission of a sensitive action, represented as a hash. 
```

Comment:
List parallelism. Personally, I would put the opening verb of this whole list in the infinitive (as the original function documentation does), i.e. "Submit a ...".
```suggestion
//! - `propose` - Submits a sensitive action, represented as a hash. 
```